docs: audit cluster management section (#485)
siderolabs/docs: public/omni/cluster-management/create-a-hybrid-cluster.mdx

description: Create a hybrid Talos cluster across networks using KubeSpan.
---

import { version } from '/snippets/custom-variables.mdx';

> A hybrid cluster is a Kubernetes cluster whose nodes span multiple networks or infrastructure types, for example, a mix of bare metal machines, cloud virtual machines, on-premises virtual machines, or single-board computers (SBCs).

Reviewer: SBC isn't a different type than bare metal.

> By default, Kubernetes assumes all nodes can reach each other directly on the same network. When nodes are spread across different networks, this assumption breaks down. <a href={`../../talos/${version}/networking/kubespan`}>Kubespan</a> addresses this by establishing an encrypted WireGuard tunnel between every node in the cluster, so that all nodes can communicate securely regardless of where they are hosted.

Reviewer: ```suggestion
Kubernetes requires all nodes can reach each other directly without NAT. When nodes are spread across different networks, this assumption breaks down. <a href={`../../talos/${version}/networking/kubespan`}>Kubespan</a> addresses this by establishing an encrypted WireGuard tunnel between every node in the cluster. The tunnel flattens the network so all nodes can communicate securely regardless of where they are hosted.
```

## Enable KubeSpan

Once your cluster is created with nodes spanning multiple networks, enable KubeSpan to allow those nodes to communicate.

KubeSpan can be enabled via a config patch, applied either through the Omni UI or a cluster template.

<Tabs>
<Tab title="Cluster Templates">

To enable KubeSpan using a cluster template, add the following patch to your cluster template definition:

```yaml
patches:
  - name: kubespan-enabled
    inline:
      machine:
        network:
          kubespan:
            enabled: true
```

For more information on patching Omni clusters inline or with patch files, see the [Cluster Template reference documentation](../reference/cluster-templates#patches).

</Tab>
<Tab title="UI">

To enable KubeSpan using the UI:

1. Navigate to your cluster in Omni.
2. Click the **...** button next to the cluster you want to patch.
3. Select **Config Patches** from the dropdown.
4. Click **Create Patch** to open the **Create Patch** page.
5. Apply the following patch:

> ```yaml

Reviewer: Talos 1.13 is going to have a multi-doc config so we might want to add tabs for that now. This config will still work but new config types are recommended.

machine:
  network:
    kubespan:
      enabled: true
```

<img src="./images/create-a-hybrid-cluster-create-patch-kubescan-enabled.png" alt="Create a new cluster-scoped config patch" />

</Tab>
</Tabs>

> Once this patch is applied, all node-to-node traffic in the cluster will be encrypted using WireGuard, allowing nodes to communicate with each other securely regardless of which network they are on.

Reviewer: We should add a warning about network throughput with wireguard is significantly less than without wireguard. If people want native network throughput from nodes on the same network they need to set up the `filters.excludeAdvertisedNetworks` configuration
